### PR TITLE
Run EndToEnd tests on circle, too.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: scripts/tests.sh [hunk 1]
```diff
@@ -30,6 +30,20 @@ set -e
...
+    if [ -z "$2" ]
+    then
+        echo "Usage: $0 [--junit_report <report_directory>]"
+        exit 1
+    fi
+    testargs_no_opt="--logger=JUNIT,test_suite,$2/no_opt.xml"
+    testargs_opt="--logger=JUNIT,test_suite,$2/opt.xml"
+else
+    testargs_no_opt=''
+    testargs_opt=''
```
Comment (axic) [inline scripts/tests.sh:44]: Any reason these are using single quotes as opposed to double quotes for the rest?

File: scripts/tests.sh [hunk 2]
```diff
@@ -30,6 +30,20 @@ set -e
...
 
+if [ "$1" -e --junit_report ]
+then
+    if [ -z "$2" ]
+    then
+        echo "Usage: $0 [--junit_report <report_directory>]"
+        exit 1
+    fi
+    testargs_no_opt="--logger=JUNIT,test_suite,$2/no_opt.xml"
+    testargs_opt="--logger=JUNIT,test_suite,$2/opt.xml"
```
Comment (axic) [inline scripts/tests.sh:41]: What is `--logger==JUNIT`? Is it useful with circleci?

File: scripts/tests.sh [hunk 3]
```diff
@@ -43,8 +57,8 @@ elif [ -z $CI ]; then
 else
     mkdir -p /tmp/test
     # Update hash below if binary is changed.
-    wget -q -O /tmp/test/eth https://github.com/ethereum/cpp-ethereum/releases/download/solidityTester/eth_byzantium2
-    test "$(shasum /tmp/test/eth)" = "4dc3f208475f622be7c8e53bee720e14cd254c6f  /tmp/test/eth"
+    wget -q -O /tmp/test/eth https://github.com/ethereum/cpp-ethereum/releases/download/solidityTester/eth_no_microhttpd
+    test "$(shasum /tmp/test/eth)" = "8e0e2c2b9f4cf4ad8956a81c1520bbb817768a53  /tmp/test/eth"
```
Comment (axic) [inline scripts/tests.sh]: Do you want to also update travis to use this newer version? It is fine testing against two different versions though.

**Review by axic — APPROVED** [review]

File: scripts/tests.sh (hunk 1 above)
Comment (chriseth) [reply scripts/tests.sh:44]: No particular.

File: scripts/tests.sh (hunk 2 above)
Comment (chriseth) [reply scripts/tests.sh:41]: Yes, it creates an XML file with test results that is parsed by circle. It then displays which exact test failed at the top of the build page.

File: scripts/tests.sh (hunk 3 above)
Comment (chriseth) [reply scripts/tests.sh]: That's the problem with this PR: The version of some library is incompatible between the two systems.

Comment (chriseth) [reply scripts/tests.sh]: The PR as it is now uses the same version for both.

Comment (axic) [reply scripts/tests.sh]: Right, for a second I thought this change is in `circle.yml` :)

**Comment by chriseth** [comment]
eth seems to either crash on circle or thinks that the database is already open. In any case, it is very unstable :(

**Comment by chriseth** [comment]
Ok, using two different versions of eth for each CI, seems to be stable now.

We might even consider disabling the endToEnd tests for travis, if we take care to run it on the relaese branch before we create the actual release tag.

**Pull request merged**